internal/core/printer.go

```diff
@@ -199,3 +206,28 @@ func (p *Printer) printJSON(data interface{}) error {
 
 	return encoder.Encode(data)
 }
+
+func (p *Printer) printYAML(data interface{}) error {
+	_, implementMarshaler := data.(yaml.Marshaler)
+	err, isError := data.(error)
+
+	if isError && !implementMarshaler {
+		data = map[string]string{
+			"error": err.Error(),
+		}
+	}
+
+	writer := p.stdout
+	if isError {
+		writer = p.stderr
+	}
+	encoder := yaml.NewEncoder(writer)
+
+	// We handle special case to make sure that a nil slice is marshal as `[]`
+	if reflect.TypeOf(data).Kind() == reflect.Slice && reflect.ValueOf(data).IsNil() {
```

Comment:
Is it needed with yaml encoding ?

Reply:
It doesn't seem to be required. I've removed it.